src/eva/vision/data/datasets/classification/camelyon16.py

```diff
@@ -0,0 +1,228 @@
+    @override
+    def classes(self) -> List[str]:
+        return ["normal", "tumor"]
+
+    @property
+    @override
+    def class_to_idx(self) -> Dict[str, int]:
+        return {"normal": 0, "tumor": 1}
+
+    @functools.cached_property
+    def annotations_test_set(self) -> Dict[str, str]:
+        """Loads the dataset labels."""
+        path = os.path.join(self._root, "testing/reference.csv")
+        reference_df = pd.read_csv(path, header=None)
+        return {k: v.lower() for k, v in reference_df[[0, 1]].itertuples(index=False)}
+
+    @override
+    def prepare_data(self) -> None:
+        _validators.check_dataset_exists(self._root, True)
+
+        expected_directories = ["training/normal", "training/tumor", "testing/images"]
+        for resource in expected_directories:
+            if not os.path.isdir(os.path.join(self._root, resource)):
+                raise FileNotFoundError(f"'{resource}' not found in the root folder.")
+
+        if not os.path.isfile(os.path.join(self._root, "testing/reference.csv")):
+            raise FileNotFoundError("'reference.csv' file not found in the testing folder.")
+
+    @override
+    def validate(self) -> None:
+
+        expected_n_files = {
+            "train": 216,
+            "val": 54,
+            "test": 129,
+            None: 399,
+        }
+        length = expected_n_files.get(self._split, 0)
+        _validators.check_number_of_files(self._file_paths, length, self._split)
+
+        _validators.check_dataset_integrity(
+            self,
+            length=None,
+            n_classes=2,
+            first_and_last_labels=("normal", "tumor"),
+        )
+
+    @override
+    def filename(self, index: int) -> str:
+        return os.path.basename(self._file_paths[self._get_dataset_idx(index)])
+
+    @override
+    def __getitem__(self, index: int) -> Tuple[np.ndarray, np.ndarray, Dict[str, Any]]:
+        return base.ImageClassification.__getitem__(self, index)
+
+    @override
+    def load_image(self, index: int) -> torch.Tensor:
+        return wsi.MultiWsiDataset.__getitem__(self, index)
+
+    @override
+    def load_target(self, index: int) -> np.ndarray:
+        file_path = self._file_paths[self._get_dataset_idx(index)]
+        return np.asarray(self._get_target_from_path(file_path))
+
+    @override
+    def load_metadata(self, index: int) -> Dict[str, Any]:
+        return {"wsi_id": self.filename(index).split(".")[0]}
+
+    def _load_file_paths(self, split: Literal["train", "val", "test"] | None = None) -> List[str]:
+        """Loads the file paths of the corresponding dataset split."""
+        train_paths, val_paths = [], []
+        for path in glob.glob(os.path.join(self._root, "training/**/*.tif")):
+            if self._get_id_from_path(path) in self._val_slides:
+                val_paths.append(path)
+            else:
+                train_paths.append(path)
+        test_paths = sorted(glob.glob(os.path.join(self._root, "testing/images", "*.tif")))
```

Comment:
I think we should apply the `sorted` only in line 213 to all the paths that are returned. Currently the sort is only applied to the test paths, so the order of the train / val paths is not deterministic